Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -2215,15 +2220,17 @@ def fetch_last_emails(
     qs = qs.filter().order_by("datetime_received")
 
     result = []
+    exclude_ids = exclude_ids if exclude_ids else set()
+    demisto.debug(f'{APP_NAME} - Exclude ID list: {exclude_ids}')
+    counter = 0
     for item in qs:
-        if isinstance(item, Message):
+        counter += 1
+        if isinstance(item, Message) and item.message_id not in exclude_ids:
             result.append(item)
             if len(result) >= client.max_fetch:
                 break
-
-    if exclude_ids and len(exclude_ids) > 0:
-        exclude_ids = set(exclude_ids)
-        result = [x for x in result if x.message_id not in exclude_ids]
+    demisto.debug(f'{APP_NAME} - Got total of {counter} from ews query. '
```

Comment:
Check if you can print also the total number of emails and not only the ones we fetched